altair/vegalite/v5/api.py

```diff
@@ -3047,7 +3061,57 @@ def interactive(
             encodings.append("x")
         if bind_y:
             encodings.append("y")
-        return self.add_params(selection_interval(bind="scales", encodings=encodings))
+        interactive_chart = self.add_params(
+            selection_interval(bind="scales", encodings=encodings)
+        ).copy()
+        # We can't simply use configure_mark since configure methods
+        # are not allowed in layered specs
+        if tooltip:
+            if isinstance(interactive_chart.mark, str):
+                interactive_chart.mark = {
+                    "type": interactive_chart.mark,
+                    "tooltip": tooltip,
+                }
+            else:
+                interactive_chart.mark.tooltip = tooltip
+        if legend:
+            if not isinstance(legend, list):
+                # Set the legend to commonly used encodings by default
+                legend = [
+                    "angle",
+                    "radius",
+                    "color",
+                    "fill",
+                    "shape",
+                    "size",
+                    "stroke",
+                ]
+                # Detect common legend encodings used in the spec
+                # legend = [
+                #     enc
+                #     for enc in interactive_chart.encoding.to_dict(validate=False).keys()
+                #     if enc
+                #     in [
+                #         "angle",
+                #         "radius",
+                #         "color",
+                #         "fill",
+                #         "shape",
+                #         "size",
+                #         "stroke",
+                #     ]
+                # ]
```

Comment:
I am thinking it's more elegant if we set the legend encodings only to encodings that actually exist in the spec. However, `to_dict(validate=False)` currently does not work for channels as per https://github.com/altair-viz/altair/issues/3075.

It is possible to do something like this instead, but it seems clunky:

```py
[
    enc for enc in vars(chart.encoding)['_kwds']
    if not isinstance(chart.encoding[enc], alt.utils.schemapi.UndefinedType)
]
```